scripts/scripts.js

```diff
@@ -218,6 +218,349 @@ function preloadFile(href, as) {
   document.head.appendChild(link);
 }
 
+function checkSSGPage() {
+  const metaSku = document.querySelector('meta[name="sku"]');
+  return metaSku?.content?.trim()?.length > 0;
+}
+
+function parseProductName(productDetails) {
+  const name = productDetails.querySelector('h1')?.textContent?.trim();
+  return name;
+}
+
+function parseProductImages(productDetails) {
```

Comment:
Shouldn't most of this product-specific logic go into the product details block? Or even a separate scripts/productdata.js? I know a lot of what you have in here has a `TODO: Remove...` comment so not sure how much is permanent vs will be removed.